### PR TITLE
feat: use helix-deploy [skip ci]

### PR DESCRIPTION
**Pull request by tripodsan**
Create a helix-pages service instead of script actions.
uses `helix-deploy` to deploy an universal action that can handle all extensions. eg:

https://adobeioruntime.net/api/v1/web/tripod/helix-pages/pages@4.14.0/html?path=/hello.md&owner=trieloff&repo=soupdemo&ref=master

https://lqmig3v5eb.execute-api.us-east-1.amazonaws.com/helix-pages/pages/4.14.0/html?path=/hello.md&owner=trieloff&repo=soupdemo&ref=master

Also see discussion in https://github.com/adobe/helix-home/discussions/174#discussioncomment-184904

still missing:
- update `helix-config.yaml` accordingly
- figure out how `helix-dispatch` can be used to serve old and new format

---
An alternative approach would be to run `helix-deploy` against all scripts, eg:

```
hedy -v --name=pages/html@1.0.0 --entryfile=.hlx/build/src/html.js --deploy
hedy -v --name=pages/plain_html@1.0.0 --entryfile=.hlx/build/src/plain_html.js --deploy
...
```

or similar to today, using the sha package name:

```
hedy -v --name=abcef1234567/html --entryfile=.hlx/build/src/html.js --deploy
hedy -v --name=abcef1234567/plain_html --entryfile=.hlx/build/src/plain_html.js --deploy
...
````

or the pages version number:

```
hedy -v --name=4.13.0/html --entryfile=.hlx/build/src/html.js --deploy
hedy -v --name=4.13.0/plain_html --entryfile=.hlx/build/src/plain_html.js --deploy
...
````

### CODE REVIEW

**Comment by trieloff** [comment]
I'd very much prefer the third alternative you've offered:

> or the pages version number:
>
> ```
> hedy -v --name=4.13.0/html --entryfile=.hlx/build/src/html.js --deploy
> hedy -v --name=4.13.0/plain_html --entryfile=.hlx/build/src/plain_html.js --deploy
> ```

As this is least disruptive in terms of packaging, but still gives us the named identifiers.

We'd still need to think about what to do with branch deployments, though.

**Comment by tripodsan** [comment]
> We'd still need to think about what to do with branch deployments, though.

when we slowly want to come to a state, where helix-pages is just a service, eg: `helix-services/pages@v1`, then we can avoid branch deployments for helix pages. the versions built in branches will have the same rules as other services. also, the CI will create ci versions.

**Comment by trieloff** [comment]
> also, the CI will create ci versions.

So `hedy -v --name=ci33883/html --entryfile=.hlx/build/src/html.js --deploy` – that would work

**Comment by tripodsan** [comment]
I created now a `deploy.sh` that deploys all the scripts to the targets.
using a patched helix-dispatch, I could now render a page in AWS:
https://master--helix-pages-test--tripodsan.99productrules.com/office/styling-test.html?a=2

still todo:
- [ ] adjust the cgi-bins to universal format
- [ ] update the helix-config.yaml after deploy
- [ ] create new strain during branch deploys

File: helix-config.yaml [hunk 1]
```diff
@@ -13,4 +13,4 @@ strains:
     code: *defaultRepo
     content: https://github.com/adobe/helix-pages.git#master
     static: https://github.com/adobe/helix-pages.git/htdocs#master
-    package: helix-pages/2c6563a3208d843e7adef7cf13aeeda324c23495
+    package: tripod/pages_4.14.1
```
Comment (trieloff) [inline helix-config.yaml:16]: This probably needs to change before the final merge.

**Review by trieloff — APPROVED** [review]

**Pull request merged**